lib/internal/fs/promises.js

```diff
@@ -663,7 +680,7 @@ async function writeFile(path, data, options) {
   options = getOptions(options, { encoding: 'utf8', mode: 0o666, flag: 'w' });
   const flag = options.flag || 'w';
 
-  if (!isArrayBufferView(data)) {
+  if (!isIterable(data) && !isArrayBufferView(data)) {
```

Comment:
I could be wrong, but I feel flip the conditions would be more efficient (especially for `Buffer` objects, which are `ArrayBufferView`): 
```suggestion
  if (!isArrayBufferView(data) && !isIterable(data)) {
```